File: 단기스터디뿌셔뿌셔/c++/prog_두개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <set>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> numbers) {
    set<int> numList;
    sort(numbers.begin(), numbers.end());
    for(int i = 0; i < numbers.size(); i++){
        for(int j = i + 1; j < numbers.size(); j++){ 
            int num = numbers[i] + numbers[j];
            numList.insert(num);
        }
    }
    vector<int> answer(numList.begin(), numList.end());
    return answer;
}

File: 단기스터디뿌셔뿌셔/java/prog_두개뽑아서 더하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] numbers) {
        Set<Integer> set = new TreeSet<Integer>(); //오름차순 정렬 set 생성(중복허용x)
        Arrays.sort(numbers); //오름차순 정렬
        
        for(int i = 0; i < numbers.length; i++){
            for(int j = i + 1; j < numbers.length; j++){
                set.add(numbers[i] + numbers[j]);
            }
        }
        // Set -> 배열 
        return set.stream().mapToInt(Integer::intValue).toArray();
    }
}